backend/src/main/java/com/bakdata/conquery/models/types/ResultType.java

```diff
@@ -255,6 +299,15 @@ public String typeInfo() {
 			return this.getClass().getAnnotation(CPSType.class).id() + "[" + elementType.typeInfo() + "]";
 		}
 
+		@Override
+		public List<T> getFromResultSet(ResultSet resultSet, int columnIndex, ResultSetProcessor resultSetProcessor) throws SQLException {
+			if (elementType instanceof DateRangeT) {
+				return elementType.getFromResultSetAsList(resultSet, columnIndex, resultSetProcessor);
```

Comment:
das ist eine etwas ungünstige abstraktion, lass mal bitte Dienstag darüber reden @thoniTUB vlt fällt dir was ein?

Ich kann aber verstehen, dass es schwierig ist hier eine sinnvolle abstraktion in der Interaktion mit den SQL Servern zu finden.

Reply:
Ich schlage vor die Methoden als Funktionen in den `ResultSetProcessor` aus zu lagern und dort eine Jump-Table/Map `ResultType`->`getFromResultSet` zu erstellen, wobei `getFromResultSet` letztendlich entfällt weil dort dierekt die Methoden aus dem `ResultSetProcessor` (`getDate`, ...) hinterlegt werden können.

Reply:
Ich hab vorher mit @jnsrnhld gesprochen, wir belassen es erstmal so, weil Listen in der implementierung ziemlich speziell sind und sich die Abstraktion ggf dann löst, wenn sie die anderen Selects auch implementieren.